nifi-nar-bundles/nifi-jolt-record-bundle/nifi-jolt-record-processors/src/main/java/org/apache/nifi/processors/jolt/record/JoltTransformRecord.java

```diff
@@ -234,7 +235,7 @@
     protected Collection<ValidationResult> customValidate(ValidationContext validationContext) {
         final List<ValidationResult> results = new ArrayList<>(super.customValidate(validationContext));
         final String transform = validationContext.getProperty(JOLT_TRANSFORM).getValue();
-        final String customTransform = validationContext.getProperty(CUSTOM_CLASS).getValue();
+        final String customTransform = validationContext.getProperty(CUSTOM_CLASS).evaluateAttributeExpressions().getValue();
```

Comment:
If you change this property to accept FlowFile attributes, you won't be able to use it the way it has been used in customValidate(), instead you'd need to check if Expression Language is present and assume it is valid in that case, then check again at runtime. If you revert it back to Variable Registry, this change can remain.

Reply:
Updated logic for variable registry.